packages/core/src/domain/errorCollection.ts

```diff
@@ -118,26 +150,39 @@ export function stopRuntimeErrorTracking() {
   ;(report.unsubscribe as (handler: Handler) => void)(traceKitReportHandler)
 }
 
-export function formatRuntimeError(stackTrace: StackTrace, errorObject: any) {
-  let message: string
-  let stack: string
-  if (stackTrace.message === undefined && !(errorObject instanceof Error)) {
-    message = `Uncaught ${jsonStringify(errorObject)}`
-    stack = 'No stack, consider using an instance of Error'
-  } else {
-    message = stackTrace.message || 'Empty message'
-    stack = toStackTraceString(stackTrace)
+export function makeAddError(errorObservable: ErrorObservable) {
+  return (addedError: AddedError, savedGlobalContext?: Context) => {
+    const stackTrace = addedError.error instanceof Error ? computeStackTrace(addedError.error) : undefined
+    const { message, stack, kind } = formatUnknownError(stackTrace, addedError.error, 'Captured')
+    errorObservable.notify({
+      message,
+      savedGlobalContext,
+      context: {
+        error: {
+          kind,
+          stack,
+          origin: addedError.source,
+        },
+      },
+      customerContext: addedError.context,
+      startTime: addedError.startTime,
+    })
+  }
+}
```

Comment:
two concerns about having this logic in shared error collection:
- since it is a rum feature, any reason to have it here? we could just export the logic that process the error and keep this component about automatic error collection only
- FMU, in a npm setup, error collection instance is shared between rum and logs, so rum added error could be retrieved by the default logger

Reply:
> since it is a rum feature, any reason to have it here? we could just export the logic that process the error and keep this component about automatic error collection only

It depends if we want to use the error frequency limit in the same way as other errors. If we move the code in RUM, it would be a bit more difficult to reuse the same limit strategy.

> FMU, in a npm setup, error collection instance is shared between rum and logs, so rum added error could be retrieved by the default logger

Isn't this an expected behavior if the log `forwardErrorsToLog` option is enabled? But you are right, this won't work on non-NPM setups.

Reply:
interesting questions!
About limit strategy, I'd say that the similar manual usage on the logs side (logger.error) is not limited so I'd be fine to keep the limiting strategy to automatic error collection for now. Maybe at some point adapting the limitation strategy if needed.
About `forwardErrorToLogs`, IMO it only concerns automatic error collection so I'd keep the two separated. It could worth discussing this point with @hdelaby though.

Reply:
My 2 cents: it's better to send the RUM errors with logs (after all they're front-end errors too) and then users will filter them in their log pipelines rather than having to use 2 APIs side by side if you want to have the log both in logs AND rum.

Reply:
it sounds fair.
since we also have a task to send log errors in RUM (RUMF-741), I'd be in favor of either:

- not send RUM errors with logs for now but handle it in RUMF-741
- tackle RUMF-741 now

IMO, tackling both at the same time could help us to handle it properly

Reply:
Agreed. I'll move the addError code in RUM then!

Reply:
Nice, I'll update the the backlog task accordingly